TriangleTypeDefinition

dzmitri-litvinov opened the pull request:
2. ТРЕУГОЛЬНИК. Заданы три числа. Выясните,  какого вида треугольник с такими длинами сторон: равносторонний, прямоугольный, равнобедренный или общего вида.

dzmitri-litvinov committed:
TriangleTypeDefinition

2. ТРЕУГОЛЬНИК. Заданы три числа. Выясните,  какого вида треугольник с такими длинами сторон: равносторонний, прямоугольный, равнобедренный или общего вида.

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+void sideInput(int* side)
+{
+	cin >> *side;
+}
```
Good example of function!

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	cin >> *side;
+}
+
+void triangleInput(int *side1, int *side2, int *side3)
+{
+	cout << "Enter the 1st side length: ";
+	sideInput(side1);
+	cout << "Enter the 2nd side length: ";
+	sideInput(side2);
+	cout << "Enter the 3rd side length: ";
+	sideInput(side3);
+}
```
Bad example of a function. Try to think please.
I mentioned such an example in the last lesson.

dzmitri-litvinov replied:
Is it possible to change in such ways?
1. Split this function into 3 functions like "side1Input", "side2Input", "side3Input". So we will have 3 functions to be used in main.
2. Just use sideInput function 3 times in main.

VadzimAlkhimionak replied:
2 variant

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	sideInput(side3);
+}
+
+void makeSide1Longest(int* side1, int* side2, int* side3)
+{
+	swapSidesIfRequired(side1, side2);
+	swapSidesIfRequired(side1, side3);
+}
```
We will discuss ше tomorrow

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+{
+	int temp;
+
+	if ((*side1) < (*side2))
```
Useless brackets ` if (*side1 < *side2)` enough

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+		return true;
+	}
+
+	return false;
```
The same `return side1 > (side2 + side3)`

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	{
+		return true;
+	}
+	else { return false; }
```
The same `return side1 == side2 and side2 == side3`

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	{
+		return true;
+	}
+	else { return false; }
```
the same example above

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	{
+		return true;
+	}
+	else { return false; }
```
the same example above

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+	{
+		return true;
+	}
+	else { return false; }
```
the same example above

VadzimAlkhimionak commented on TriangleType.cpp:
```diff
@@ -0,0 +1,148 @@
+
+void triangleTypeDefinition(int side1, int side2, int side3)
+{
+	if (isTriangleEquilateral(side1, side2, side3))
+	{
+		cout << endl << "The triangle is equilateral." << endl;
+	}
```
not suitable `if`  for this function

dzmitri-litvinov replied:
To tell the truth, I don't understand why it is not suitable. Where can I read about it?

VadzimAlkhimionak reviewed (commented):
Some comments for you :)

dzmitri-litvinov committed:
Update TriangleType.cpp

3 comments are not completed. Let's discuss it during the next meeting or before

dzmitri-litvinov committed:
Update TriangleType.cpp
